### PR TITLE
CR "stardust" fork

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: seq/detail/config.h [hunk 1]
```diff
@@ -27,33 +27,35 @@ namespace seq
...
+    virtual void syncData( const uint128_t& ) { /* nop */ }
+    virtual void unmapData() { /* nop */ }
 
-        virtual bool mapData( const uint128_t& ) { return true; }
-        virtual void syncData( const uint128_t& ) { /* nop */ }
-        virtual void unmapData() { /* nop */ }
+    co::Object* getInitData();
 
-        co::Object* getInitData();
+    virtual ObjectMap* getObjectMap();
```
Comment (tribal-tec) [inline seq/detail/config.h]: virtual is needed?

File: seq/application.cpp [hunk 2]
```diff
@@ -60,6 +62,24 @@ eq::Config* Application::getConfig()
     return _impl->getConfig();
 }
 
+bool Application::registerObject( co::Object* object, const uint32_t type )
+{
+    if( !_impl || !_impl->getConfig() )
```
Comment (tribal-tec) [inline seq/application.cpp]: wrong spacing in parentheses

File: seq/application.h [hunk 3]
```diff
@@ -136,6 +137,30 @@ class Application : public eq::Client, public co::ObjectFactory
...
+     * @version 1.8
+     * @sa co::ObjectMap::register_()
+     */
+    SEQ_API bool registerObject( co::Object* object, const uint32_t type );
+
+    /**
+     * Remove and deregister an object.
+     *
+     * @param object the object to remove and deregister
+     * @return false on if object was not registered, true otherwise
```
Comment (tribal-tec) [inline seq/application.h]: remove 'on'

File: seq/detail/window.h [hunk 4]
```diff
@@ -25,34 +26,34 @@ namespace seq
...
+    virtual void frameFinish( const uint128_t& frameID,
+                              const uint32_t frameNumber );
+    virtual bool configInitGL( const uint128_t& initID );
+    virtual bool configExitGL();
+
+    bool initContext() { return eq::Window::configInitGL( uint128_t( )); }
+    bool exitContext() { return eq::Window::configExitGL(); }
+    //@}
+
+    virtual bool processEvent( const eq::Event& event );
```
Comment (tribal-tec) [inline seq/detail/window.h]: override/final

File: seq/detail/window.h [hunk 5]
```diff
@@ -25,34 +26,34 @@ namespace seq
...
+    Window( eq::Pipe* parent );
+
+    Config* getConfig();
+    Pipe* getPipe();
+    seq::Renderer* getRenderer();
+    detail::Renderer* getRendererImpl();
+
+    /** @name Operations. */
+    //@{
+    virtual void frameStart( const uint128_t& frameID,
```
Comment (tribal-tec) [inline seq/detail/window.h]: override/final

File: seq/renderer.h [hunk 6]
```diff
@@ -210,11 +232,36 @@ class Renderer : public co::ObjectFactory
...
+    /**
+     * Map and return an object.
+     *
+     * @param identifier unique object identifier used for map operation
+     * @param instance already created instance to skip factory creation
+     * @return 0 if not registered, the valid instance otherwise
+     * @version 1.8
+     * @sa co::ObjectMap::map()
+     */
+    SEQ_API co::Object*	mapObject( const uint128_t& identifier,
```
Comment (bilgili) [inline seq/renderer.h]: spacing problem ?

Comment (eile) [inline seq/renderer.h]: imported tabs :(

**Comment by eile** [comment]
all done.

**Comment by bilgili** [comment]
+1

File: seq/detail/config.h [hunk 7]
```diff
@@ -27,33 +27,34 @@ namespace seq
...
 
-        seq::Application* getApplication();
-        detail::Application* getApplicationImpl();
+    seq::Application* getApplication();
+    detail::Application* getApplicationImpl();
 
-        virtual bool init() { LBDONTCALL; return false; }
-        virtual bool run( co::Object* ) { LBDONTCALL; return false; }
-        virtual bool exit() { LBDONTCALL; return false; }
+    virtual bool init() { LBDONTCALL; return false; }
```
Comment (tribal-tec) [inline seq/detail/config.h]: override/final

**Comment by tribal-tec** [comment]
+1 otherwise

**Pull request merged**